### PR TITLE
Fix image Jenkinx X build failures

### LINKED ISSUE #2443: CI builds sklearn server with wrong sklearn version -> TestNotebooks.test_explainer_examples fails
The `seldonio/sklearnserver_rest:1.2.3-dev` build by CI 
```
$ docker run --rm -it seldonio/sklearnserver_rest:1.2.3-dev bash
(base) [root@a38e7265b896 microservice]# cat requirements.txt 
seldon_core
scikit-learn == 0.20.3
numpy >= 1.8.2
joblib >= 0.13.0
```
whereas currently on master it should be
```
seldon_core
scikit-learn == 0.23.2
numpy >= 1.8.2
joblib >= 0.13.0
```

mismatch of sklearn versions causes `TestNotebooks.test_explainer_examples` test to fail

### PR DESCRIPTION
**Pull request by axsaucedo**
Fixes #2443 

The jenkins X uses the VERSION branch as the base branch, so when it was changed to used `nighly-<version>` it broke the image build, resulting in the wrong branch being built. Now this builds on the correct branch back to `v<version>`

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu Sep 17 12:58:01 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2449/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2449 --build=2

**Comment by seldondev** [comment]
Thu Sep 17 12:58:01 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2449/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2449 --build=1

**Comment by RafalSkolasinski** [comment]
Should also fix https://github.com/SeldonIO/seldon-core/pull/2449

**Comment by RafalSkolasinski** [comment]
Should also fix https://github.com/SeldonIO/seldon-core/issues/2444

**Review by RafalSkolasinski — APPROVED** [review]
/lgtm

let's merge it and see if all problems go away!

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/2449#issuecomment-694225290 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-694225290/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu Sep 17 13:21:03 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2449/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2449 --build=3

**Comment by seldondev** [comment]
Thu Sep 17 13:21:08 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2449/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2449 --build=4

**Pull request merged**
